test/ssttpcalculator/legacy/LegacyCalculatorServiceSpec.scala

```diff
@@ -44,207 +44,145 @@ class LegacyCalculatorServiceSpec extends ItSpec with Matchers with DateSupport
   private val initialPayment = BigDecimal(debt - minimumBalanceAfterInitialPayment)
   private val initialPaymentTooLarge = BigDecimal(468.01)
 
-  val calculatorService: CalculatorService = fakeApplication().injector.instanceOf[CalculatorService]
+  val calculatorService: CalculatorService = app.injector.instanceOf[CalculatorService]
 
   private def clockForMay(dayInMay: Int) = {
     val formattedDay = "%02d".format(dayInMay)
     val currentDateTime = LocalDateTime.parse(s"${_2020}-05-${formattedDay}T00:00:00.880").toInstant(UTC)
     Clock.fixed(currentDateTime, systemDefault)
   }
 
+  def addWorkingDaysResult(clock: Clock, result: LocalDate) = {
+    val today = LocalDate.now(clock)
+    AddWorkingDaysResult(today, 5, result)
+  }
+
   "payTodayRequest should" - {
     "return a payment schedule request when" - {
       "the current date is the 1st" in {
         val clock = clockForMay(_1st)
 
-        makeCalculatorInputForPayToday(debits)(clock) shouldBe TaxPaymentPlan(
+        makeCalculatorInputForPayToday(debits, addWorkingDaysResult(clock, may(_11th)))(clock) shouldBe TaxPaymentPlan(
           debits, noInitialPayment, startDate = may(_1st), endDate = may(_31st), firstPaymentDate = Some(june(_1st)))
       }
     }
 
     "the current date is the 28th" in {
       val clock = clockForMay(_28th)
 
-      makeCalculatorInputForPayToday(debits)(clock) shouldBe TaxPaymentPlan(
+      makeCalculatorInputForPayToday(debits, addWorkingDaysResult(clock, june(_7th)))(clock) shouldBe TaxPaymentPlan(
         debits, noInitialPayment, startDate = may(_28th), endDate = june(_27th), firstPaymentDate = Some(june(_28th)))
     }
 
     "the current date is the 29th" in {
       val clock = clockForMay(_29th)
 
-      makeCalculatorInputForPayToday(debits)(clock) shouldBe TaxPaymentPlan(
+      makeCalculatorInputForPayToday(debits, addWorkingDaysResult(clock, june(_8th)))(clock) shouldBe TaxPaymentPlan(
         debits, noInitialPayment, startDate = may(_29th), endDate = june(_30th), firstPaymentDate = Some(july(_1st)))
     }
   }
 
   "paymentScheduleRequest should" - {
     "return a payment schedule request without an initial payment when" - {
-      "the current date is Friday 1st May with upcoming bank holiday" in {
```

Comment:
we don't handle the working day calculation directly in this service anymore so I removed a bunch of tests around bank holiday stuff